app/src/main/java/com/capstone/weatherapp/CityListAdapter.kt

```diff
@@ -28,9 +30,14 @@ class CityListAdapter (private var cityList : List<City> = emptyList()) : Recycl
     override fun onBindViewHolder(holder: CityListAdapter.ViewHolder, position: Int) {
         holder.binding.cityName.text = cityList[position].name
         holder.binding.country.text = cityList[position].sys.country
-        holder.binding.temp.text = "${cityList[position].main.temp}℉"
-        holder.binding.lowHigh.text = "${cityList[position].main.temp_min}℉/${cityList[position].main.temp_max}℉"
-        holder.binding.humidity.text = "${cityList[position].main.humidity}%"
+        holder.binding.temp.text = String.format("%.0f℉", cityList[position].main.temp)
+        holder.binding.lowHigh.text = String.format("%.0f℉/%.0f℉", cityList[position].main.temp_min, cityList[position].main.temp_max)
+        holder.binding.humidity.text = String.format("%.0f%%", cityList[position].main.humidity)
+
+        holder.binding.root.setOnClickListener {
```

Comment:
instead of adding a click listener to binding.root, give your <CardView in city_item.xml an id and set the clickListener to that CardView object